### PR TITLE
Notifier web api

### LINKED ISSUE #23: Email Notifier and Web API
I think I was assigned these two in our last meeting.

Two questions:

1. Do we store a list of emails for each user, or do we store an email for each job?
2. Do we need HTTPS for the server or plain HTTP?

### PR DESCRIPTION
**Pull request by ammen99**
Fixes #23

### CODE REVIEW

File: src/ja/common/job.py [hunk 1]
```diff
@@ -63,12 +63,14 @@ class Job:
     """!
     Represents a job with all of its attributes.
     @param owner_id The unix user id of the user who owns this job.
```
Comment (nikolatzotchev) [inline src/ja/common/job.py]: unix -> Unix

Comment (ammen99) [reply src/ja/common/job.py]: This is not a part of this PR at all. Send another pull request to fix it :)

File: src/ja/server/web_api.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+class StatisticsWebServer:
+    """
+    Creates a web server which enables external applications to obtain statistics about JobAdder.
+    """
+
+    def __init__(self, host: str, port: int, database: ServerDatabase):
+        """!
+        Initialize the web server.
+
+        @param host The host for the server.
```
Comment (JohannesGaessler) [inline src/ja/server/web_api.py]: What do you mean by host?

Comment (ammen99) [reply src/ja/server/web_api.py]: Sorry, I haven't properly checked the documentation, it is called server_name: https://docs.python.org/3/library/http.server.html#http.server.HTTPServer

**Comment by ammen99** [comment]
I have imagined that we're going to have different classes for each request to the web server, just like we have different classes for commands, so I have added a class for each command. The list of requests is taken from https://github.com/DistributedTaskScheduling/documentation/pull/82

File: src/ja/server/web/requests.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+class WorkMachineWorkloadRequest(WebRequest):
+    """
+    Generates the response to the machine workload request.
+    """
+
+    def __init__(self, uid: str):
+        """!
+        Initialize the request response.
+
+        @param uid The uid of the work machine this request is for.
```
Comment (JohannesGaessler) [inline src/ja/server/web/requests.py]: Can this be None to get the workload of all machines?

File: src/ja/server/web/requests.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+class JobInformationRequest(WebRequest):
+    """
+    Generates the response to the job information request.
+    """
+
+    def __init__(self, uid: str):
+        """!
+        Initialize the request response.
+
+        @param uid The uid of the job the report is for.
```
Comment (JohannesGaessler) [inline src/ja/server/web/requests.py]: Can this be None to get a report for all jobs?

File: src/ja/server/web/requests.py (hunk 3 above)
Comment (ammen99) [reply src/ja/server/web/requests.py]: I don't think so.

File: src/ja/server/web/requests.py (hunk 4 above)
Comment (ammen99) [reply src/ja/server/web/requests.py]: Again, I don't think so.

File: src/ja/server/web/requests.py (hunk 3 above)
Comment (ammen99) [reply src/ja/server/web/requests.py]: You can use the workload request to get the IDs of all machines, and then you can issue a request for each machine to get the running jobs, if that's what the user needs.

**Pull request merged**